plugin/cmd/kube-scheduler/app/configurator.go

```diff
@@ -186,3 +191,100 @@ func (sc schedulerConfigurator) Create() (*scheduler.Config, error) {
 
 	return sc.CreateFromConfig(*policy)
 }
+
+func (sc *schedulerConfigurator) SetupPolicyConfigMapEventHandlers(client clientset.Interface, informerFactory informers.SharedInformerFactory) {
+	// selector targets only the scheduler's policy ConfigMap.
+	selector := cache.NewListWatchFromClient(client.CoreV1().RESTClient(), "configmaps", sc.policyConfigMapNamespace, fields.OneTermEqualSelector(api.ObjectNameField, string(sc.policyConfigMap)))
```

Comment:
it seems very odd to go to all the trouble to make the scheduler watch the API, only as a signal to kill itself. why not just poll the specified config file if all it is doing is looking for changes in order to exit?